conn.go

```diff
@@ -222,6 +219,18 @@ func (c *Conn) authAllowed() bool {
 	return !c.server.AuthDisabled && (isTLS || c.server.AllowInsecureAuth)
 }
 
+// handleError writes errors responses and closes the connection once too many
+// have occurred.
+func (c *Conn) handleError(code int, ec EnhancedCode, msg string) {
+	c.WriteResponse(code, ec, msg)
+
+	c.errCount++
+	if c.errCount > c.errThreshold {
+		c.WriteResponse(500, EnhancedCode{5, 5, 2}, "Too many errors. Quiting now")
```

Comment:
RFC 5321 defines a special status code to use when connection is closed - 421. Would using it here be better?

Reply:
Would something like `c.WriteResponse(421, EnhancedCode{5, 7, 0}, "Too many errors. Quiting now")` work? Sort of stabbing about for that enhanced code. Persistent security related issue seems the most closely related to me.

Reply:
First number in enhanced code should match the basic code, therefore it should be at least 4.7.0.
The list of registered codes can be found here: https://www.iana.org/assignments/smtp-enhanced-status-codes
I believe 4.5.1 would be the most meaningful.

Reply:
First number in enhanced code should match the basic code, therefore it should be at least 4.7.0.
The list of registered codes can be found here: https://www.iana.org/assignments/smtp-enhanced-status-codes
I believe 4.5.1 would be the most meaningful.

Reply:
[RFC 3463 Section 3.6](https://tools.ietf.org/html/rfc3463#section-3.6) Specifies that `x.5.1` is useful only as a permenant error (so 5.5.1).

Reply:
Okay, I think 421 is to be used when server closes the connection for some internal reason (e.g. shutdown). 500 5.5.1 seems fine to indicate the case of possibly malfunctioning client. I had some concerns about how permanently failing delivery (5xx) would affect such cases but I think it is fine.